src/sage/topology/moment_angle_complex.py

```diff
@@ -0,0 +1,803 @@
+            Moment-angle complex of Simplicial complex with vertex set
+            (0, 1, 2, 3) and facets {(0, 1, 2), (1, 2, 3)}
+            sage: Z._homology_group(3, base_ring=ZZ,
+            ....:                   reduced=True, verbose=False,
+            ....:                   cohomology=False, algorithm='pari')
+            Z
+            sage: Z._homology_group(4, base_ring=ZZ,
+            ....:                   reduced=True, verbose=False,
+            ....:                   cohomology=False, algorithm='pari')
+            0
+            sage: Z.homology()
+            {0: 0, 1: 0, 2: 0, 3: Z, 4: 0, 5: 0, 6: 0, 7: 0}
+            sage: RP2 = simplicial_complexes.RealProjectivePlane()
+            sage: Z = MomentAngleComplex(RP2)
+            sage: Z._homology_group(8, base_ring=ZZ,
+            ....:                   reduced=True, verbose=False,
+            ....:                   cohomology=False, algorithm='pari')
+            C2
+        """
+        if i == 0:
+            # This is a special case when computing (co)homology
+            if reduced:
+                return HomologyGroup(0, base_ring)
+            return HomologyGroup(1, base_ring)
+
+        vertices = self._simplicial_complex.vertices()
+        n = len(vertices)
+        invfac = []
+
+        for j in range(n+1):
+            for x in combinations(vertices, j):
+                S = self._simplicial_complex.generated_subcomplex(x)
+                if base_ring.is_field():
+                    invfac.append(S.homology(i-j-1, base_ring=base_ring,
+                                             cohomology=cohomology, algorithm=algorithm,
+                                             verbose=verbose, reduced=True).dimension())
+                else:
+                    invfac.extend(S.homology(i-j-1, base_ring=base_ring,
+                                             cohomology=cohomology, algorithm=algorithm,
+                                             verbose=verbose, reduced=True)._original_invts)
+
+        if base_ring.is_field():
+            return HomologyGroup(sum(invfac), base_ring)
+
+        m = len(invfac)
+        return HomologyGroup(m, base_ring, invfac)
+
+    def homology(self, dim=None, base_ring=ZZ, cohomology=False,
+                 algorithm='pari', verbose=False, reduced=True):
+        r"""
+        The (reduced) homology of ``self``.
+
+        INPUT:
+
+        - ``dim`` -- integer, or a list of integers; represents the
+          homology (or homologies) we want to compute
+        - ``base_ring`` -- commutative ring (default: ``ZZ``); must be ``ZZ``
+          or a field
+        - ``cohomology`` -- boolean (default: ``False``);
+          if ``True``, compute cohomology rather than homology
+        - ``algorithm`` -- string (default: ``'pari'``); the options are
+          ``'auto'``, ``'dhsw'``, or ``'pari'``; see below for a description
+          of what they mean
```

Comment:
This is not correct. The `algorithm` here should link to the correct place; this is not described in this method.